src/matrices/skeletonization.jl

```diff
@@ -0,0 +1,396 @@
+#=
+In this file the core computations for skeletonizing a recurrence matrix
+are defined.
+=#
+
+# Core function, which gets exported
+"""
+    skeletonize(R) → R_skel
+
+Skeletonizes the recurrence matrix `R` (see [`recurrence_matrix`](@ref)) by using
```

Comment:
Notice that the function `recurrence_matrix` is not part of the Public API, so it is not in the docs. This link will not be valid. You can use `RecurrenceMatrix` instead, but I think no need to link anything here.